modAionImpl/src/org/aion/zero/impl/sync/ResponseMgr.java

```diff
@@ -0,0 +1,98 @@
+package org.aion.zero.impl.sync;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class ResponseMgr {
+
+    private final Lock responseLock = new ReentrantLock();
+
+    /**
+     * Records time of requests/responses for active peer nodes
+     *
+     * @implNote Access to this resource is managed by the {@Link #responseLock}.
+     */
+    private final Map<String, LinkedList<Long>> requestTimeByPeers = new HashMap<>();
```

Comment:
a queue would be a better data structure here since we always add at the end and remove from the beginning